### DISCUSSION TITLE
Alternative way to get exported image byte

### DISCUSSION THREAD

**Discussion by Chris2137** [hm21/pro_image_editor#57] [Q&A] [answered]
Hi @hm21 ,

  To refresh the memory, https://github.com/hm21/pro_image_editor/issues/50#issuecomment-2120062981, I wanted to use our own "Confirm" and "Cancel" button in the dialog to save edited image. But the only available function 'doneEditing' in main_editor is not returning the byte that I need. 

  I tried to add some print out message in onImageEditingComplete, before and after calling doneEditing, when the dialog is closed (called from caller). I found that onImageEditingComplete is called last, means the dialog has already been closed before the edited bytes are returned.

  Would you have any suggestion on how can I get the edited bytes when I close the dialog? (I need to show the updated image in the parent page, so it is best to have the edited byte refreshed before actually closing the dialog).

Thanks!

**Comment by hm21** [comment, marked as answer]
Hi Chris, 

There are two ways to do this like below. I recommend option 2.

## Option 1
You can call the `doneEditing` function and then use the callbacks in the following way. But keep in mind that the editor will not work after you call `doneEditing`.

```dart
callbacks: ProImageEditorCallbacks(
  onImageEditingComplete: (bytes) async {
    // Handle the bytes, but don't call Navigator.pop(context)
  },
  onCloseEditor: () {
    // Empty function, but requires that the editor doesn't use its own logic
  },
),
```

## Option 2
I've just released version 3.0.0-dev.17! Now, you can call `captureEditorImage` to retrieve the current image bytes without closing the editor or triggering any loading dialogs.

```dart
Uint8List bytes = await editorKey.currentState!.captureEditorImage();
```

**Reply by Chris2137** [reply]
Lovely. Million thanks @hm21 , I'll try that out.
